addons/google_drive/models/google_drive.py

```diff
@@ -0,0 +1,185 @@
+        else:
+            self.model = False
+            self.filter_id = False
+
+    @api.one
+    def get_google_drive_url(self, res_id, template_id):
+        record = self.env[self.model_id.model].browse(res_id).read()[0]
+        record.update({'model': self.model_id.name, 'filter': self.filter_id.name})
+        name_gdocs = self.name_template
+        try:
+            name_gdocs = name_gdocs % record
+        except:
+            raise UserError(_("At least one key cannot be found in your Google Drive name pattern"))
+        ir_attachment = self.env['ir.attachment'].search([('res_model', '=', self.model_id.model), ('name', '=', name_gdocs),
+                                                          ('res_id', '=', res_id)], limit=1)
+        if ir_attachment:
+            return ir_attachment.url
+        else:
+            return self.copy_doc(res_id, template_id, name_gdocs, self.model_id.model).get('url')
+
+    @api.model
+    def get_access_token(self, scope=None):
+        IrConfigParameter = self.env['ir.config_parameter']
+        google_drive_refresh_token = IrConfigParameter.get_param('google_drive_refresh_token')
+        if not google_drive_refresh_token:
+            if self.env.user._is_admin():
+                action_id = self.env.ref('base_setup.action_general_configuration').id
+                msg = _("You haven't configured 'Authorization Code' generated from google, Please generate and configure it .")
+                raise RedirectWarning(msg, action_id, _('Go to the configuration panel'))
+            else:
+                raise UserError(_("Google Drive is not yet configured. Please contact your administrator."))
+        google_drive_client_id = IrConfigParameter.get_param('google_drive_client_id')
+        google_drive_client_secret = IrConfigParameter.get_param('google_drive_client_secret')
+        #For Getting New Access Token With help of old Refresh Token
+
+        data = werkzeug.url_encode(dict(client_id=google_drive_client_id,
+                                        refresh_token=google_drive_refresh_token,
+                                        client_secret=google_drive_client_secret,
+                                        grant_type="refresh_token",
+                                        scope=scope or 'https://www.googleapis.com/auth/drive'))
+        headers = {"Content-type": "application/x-www-form-urlencoded"}
+        try:
+            req = urllib2.Request('https://accounts.google.com/o/oauth2/token', data, headers)
+            content = urllib2.urlopen(req, timeout=TIMEOUT).read()
+        except urllib2.HTTPError:
+            if self.env.user._is_admin():
+                action_id = self.env.ref('base_setup.action_general_configuration').id
+                msg = _("Something went wrong during the token generation. Please request again an authorization code .")
+                raise RedirectWarning(msg, action_id, _('Go to the configuration panel'))
```

Comment:
We can directly provide message in RedirectWarning(msg is used once), applies to whole file